docker/jenkins/unit.sh

```diff
@@ -23,10 +23,8 @@ function cleanup {
     docker-compose -f $COMPOSE_FILE_LOC \
                    -p $COMPOSE_PROJECT_NAME \
                    down --remove-orphans
-    docker ps -a --no-trunc  | grep $COMPOSE_PROJECT_NAME \
-        | awk '{print $1}' | xargs --no-run-if-empty docker stop
-    docker ps -a --no-trunc  | grep $COMPOSE_PROJECT_NAME \
-        | awk '{print $1}' | xargs --no-run-if-empty docker rm
+    # Untag LB images that were built before this test run
+    docker image rm $(docker images --filter="before=$COMPOSE_PROJECT_NAME_listenbrainz" --filter "label=org.label-schema.name=ListenBrainz" --format '{{.Repository}}:{{.Tag}}')
```

Comment:
This seems like a clever approach, but I worry that it is a bit too simplistic. From what I am reading, two successive test runs of the same project will have the same filter key, which could cause it to not remove enough containers. Which is far far better than what we had before and eventually will stabilize itself. But if we were to only ever test one project, I am not sure this would work.

Reply:
`COMPOSE_PROJECT_NAME` includes `${BUILD_TAG}`, which is the incrementing build number in jenkins. Every successive test run will have a different argument to `before`.

This means that if we are running job 3, this will select items with label `name=ListenBrainz` (the listenbrainz image) and built before job 3, this would include job 2 (and job 1 if it hasn't yet been deleted).

Note that we're not using this to say "delete every other project in jenkins that was previously built", we're only saying "delete old versions of the LB run".

Reply:
OK, I didn't understand the nuances of this, let's give it a shot.